### PR TITLE
feat(cdsctl): allow manual run a pipeline not launched

### PR DESCRIPTION
**Pull request by yesnault**
Signed-off-by: Yvonnick Esnault <yvonnick.esnault@corp.ovh.com>

1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: cli/cdsctl/workflow_run.go [hunk 1]
```diff
@@ -175,14 +175,14 @@ func workflowRunManualRun(v cli.Values) error {
...
-			for _, wnr := range wnrs {
-				wn := wr.Workflow.WorkflowData.NodeByID(wnr.WorkflowNodeID)
-				if wn.Name == v.GetString("node-name") {
-					fromNodeID = wnr.WorkflowNodeID
-					break
-				}
+		wr.Workflow.VisitNode(func(node *sdk.Node, wf *sdk.Workflow) {
+			if node.Name == v.GetString("node-name") {
+				fromNodeID = node.ID
 			}
```
Comment (richardlt) [inline cli/cdsctl/workflow_run.go:181]: Can we break this loop ?

File: cli/cdsctl/workflow_run.go [hunk 2]
```diff
@@ -175,14 +175,14 @@ func workflowRunManualRun(v cli.Values) error {
...
-		for _, wnrs := range wr.WorkflowNodeRuns {
-			for _, wnr := range wnrs {
-				wn := wr.Workflow.WorkflowData.NodeByID(wnr.WorkflowNodeID)
-				if wn.Name == v.GetString("node-name") {
-					fromNodeID = wnr.WorkflowNodeID
-					break
-				}
+		wr.Workflow.VisitNode(func(node *sdk.Node, wf *sdk.Workflow) {
+			if node.Name == v.GetString("node-name") {
+				fromNodeID = node.ID
```
Comment (fsamin) [inline cli/cdsctl/workflow_run.go:180]: be careful, fromNodeId is not a pointer and you are scoped in a closure

**Review by fsamin — CHANGES_REQUESTED** [review]

**Review by sguiheux — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14909.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

File: cli/cdsctl/workflow_run.go (hunk 1 above)
Comment (yesnault) [reply cli/cdsctl/workflow_run.go:182]: fixed

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14911.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Review by fsamin — APPROVED** [review]

**Review by richardlt — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14928.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✘  
  * Standalone Tests ✘

**Pull request merged**